disassemblers/ofrak_pyghidra/src/ofrak_pyghidra/components/pyghidra_components.py

```diff
@@ -164,32 +164,27 @@ class PyGhidraDecompilationAnalyzer(CachedDecompilationAnalyzer):
 
     async def analyze(self, resource: Resource, config=None):
         program_r = await resource.get_only_ancestor(ResourceFilter.with_tags(PyGhidraProject))
-        if not self.analysis_store.get_analysis(program_r.get_id())["metadata"]["decompiled"]:
-            with TemporaryDirectory() as tempdir:
-                program_file = os.path.join(tempdir, "program")
-                await program_r.flush_data_to_disk(program_file)
-
-                try:
-                    program_attrs = program_r.get_attributes(ProgramAttributes)
-                except NotFoundError:
-                    program_attrs = await program_r.analyze(ProgramAttributes)
-
-                base_address = None
-                if not any(program_r.has_tag(tag) for tag in _GHIDRA_AUTO_LOADABLE_FORMATS):
-                    regions = await program_r.get_children_as_view(
-                        MemoryRegion, r_filter=ResourceFilter.with_tags(MemoryRegion)
-                    )
-                    base_address = min(r.virtual_address for r in regions)
-
-                self.analysis_store.store_analysis(
-                    program_r.get_id(),
-                    unpack(
-                        program_file,
-                        True,
-                        language=_arch_info_to_processor_id(program_attrs),
-                        base_address=base_address,
-                    ),
-                )
+        complex_block = await resource.view_as(ComplexBlock)
+        cb_key = f"func_{complex_block.virtual_address}"
+        if self.analysis_store.id_exists(program_r.get_id()):
+            analysis = self.analysis_store.get_analysis(program_r.get_id())
+        else:
+            tempdir = mkdtemp(prefix="rbs-pyghidra-bin")
+            program_file = os.path.join(tempdir, "program")
+            await program_r.flush_data_to_disk(program_file)
+            try:
+                program_attrs = program_r.get_attributes(ProgramAttributes)
+            except NotFoundError:
+                program_attrs = await program_r.analyze(ProgramAttributes)
+            analysis = unpack(
+                program_file, False, language=_arch_info_to_processor_id(program_attrs)
+            )
+        if "decompilation" not in analysis[cb_key]:
+            program_file = analysis["metadata"]["path"]
+            await program_r.flush_data_to_disk(program_file)
```

Comment:
Since we're changing the program file on disk during analysis, we should probably document somewhere that we're assuming the path is in a temporary directory. We don't want to end up with a situation where `unpack` or `analyze` is called on a non-temporary file, and it gets modified unexpectedly.